crates/forge/src/main.rs

```diff
@@ -70,36 +74,41 @@ fn main_execution() -> Result<bool> {
 
     let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
 
-    let build_output = Command::new("scarb")
-        .current_dir(std::env::current_dir().context("Failed to get current directory")?)
-        .arg("build")
-        .stderr(Stdio::inherit())
-        .stdout(Stdio::inherit())
-        .output()
-        .context("Failed to build contracts with Scarb")?;
-    if !build_output.status.success() {
-        bail!("Scarb build did not succeed")
-    }
+    let packages: Vec<PackageMetadata> = args
+        .packages_filter
+        .match_many(&scarb_metadata)
+        .context("Failed to find any packages matching the specified filter")?;
 
     let mut all_failed_tests = vec![];
-    for package in &scarb_metadata.workspace.members {
-        let forge_config = forge::scarb::config_from_scarb_for_package(&scarb_metadata, package)?;
-        let (package_path, lib_path) = paths_for_package(&scarb_metadata, package)?;
-
+    for package in &packages {
+        let forge_config = config_from_scarb_for_package(&scarb_metadata, &package.id)?;
+        let (package_path, lib_path) = paths_for_package(&scarb_metadata, &package.id)?;
         std::env::set_current_dir(package_path.clone())?;
 
-        let package_name = name_for_package(&scarb_metadata, package)?;
-        let dependencies = dependencies_for_package(&scarb_metadata, package)?;
-        let target_name = target_name_for_package(&scarb_metadata, package)?;
-        let corelib_path = corelib_for_package(&scarb_metadata, package)?;
+        let target_dir = target_dir_for_package(&package_path)?; // TODO(#671)
```

Comment:
Let's move the todo above the lines and include an short explanation alongside the issue number

Reply:
It is really well explained in the issue imo, will move it up though